KAFKA-8316: Remove deprecated usage of Slf4jRequestLog, SslContextFactory

dongjinleekr opened the pull request:
This PR resolves [KAFKA-8316](https://issues.apache.org/jira/browse/KAFKA-8316) following the method described [here](https://github.com/eclipse/jetty.project/issues/3502), along with [KAFKA-8308](https://issues.apache.org/jira/browse/KAFKA-8308).

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

ijuma commented:
Thanks for the PR. Can you please rebase against trunk?

ijuma commented on connect/runtime/src/main/java/org/apache/kafka/connect/runtime/rest/util/SSLUtils.java:
```diff
@@ -47,7 +47,7 @@ public static SslContextFactory createSslContextFactory(WorkerConfig config) {
         Map<String, Object> sslConfigValues = config.valuesWithPrefixAllOrNothing("listeners.https.");
 
-        SslContextFactory ssl = new SslContextFactory();
+        SslContextFactory.Server ssl = new SslContextFactory.Server();
```
I think you have to check the `client` boolean to decide which factory to use, right?

dongjinleekr commented:
@ijuma Here is the update. Since `SSLUtils#configureSslContextFactoryAuthentication` should be called for `SslContextFactory.Server` only, I splitted `SSLUtils#createSslContextFactory` into `SSLUtils#create[Server, Client]SideSslContextFactory`. How about this approach?

dongjinleekr committed:
Remove deprecated class Slf4jRequestLog: use Slf4jRequestLogWriter, CustomRequestLog instread.

1. Remove '@SuppressWarnings("deprecation")' from RestServer#initializeResources, JsonRestServer#start.
2. Remove unused JsonRestServer#httpRequest.

dongjinleekr committed:
Fix deprecated class usage: SslContextFactory -> SslContextFactory.[Server, Client]

1. Split SSLUtils#createSslContextFactory into SSLUtils#create[Server, Client]SideSslContextFactory: each method instantiates SslContextFactory.[Server, Client], respectively.
2. SSLUtils#configureSslContextFactoryAuthentication is called from SSLUtils#createServerSideSslContextFactory only.
3. Update SSLUtilsTest following splittion; for client-side SSL Context Factory, SslContextFactory#get[Need, Want]ClientAuth is always false. (SSLUtilsTest#testCreateClientSideSslContextFactory)

dongjinleekr commented:
Retest this please.

dongjinleekr commented:
Retest this please.

dongjinleekr commented:
Rebased onto the latest trunk. cc/ @ijuma @rhauch @C0urante

hachikuji commented on tools/src/main/java/org/apache/kafka/trogdor/rest/JsonRestServer.java:
```diff
@@ -101,10 +102,9 @@ public void start(Object... resources) {
-        requestLog.setLogLatency(true);
+        Slf4jRequestLogWriter slf4jRequestLogWriter = new Slf4jRequestLogWriter();
+        slf4jRequestLogWriter.setLoggerName(JsonRestServer.class.getCanonicalName());
+        CustomRequestLog requestLog = new CustomRequestLog(slf4jRequestLogWriter, CustomRequestLog.EXTENDED_NCSA_FORMAT + " %msT");
```
How did you come up with this format string? I could be mistaken, but I think the old implementation was just `CustomRequestLog.EXTENDED_NCSA_FORMAT` without `%msT`.

dongjinleekr replied:
`" %msT"` is for `requestLog.setLogLatency(true);`. As you can see from the code of [AbstractNCSARequestLog](https://github.com/eclipse/jetty.project/blob/jetty-9.4.14.v20181114/jetty-server/src/main/java/org/eclipse/jetty/server/AbstractNCSARequestLog.java#L205) which is extended by [Slf4jRequestLog](https://github.com/eclipse/jetty.project/blob/jetty-9.4.14.v20181114/jetty-server/src/main/java/org/eclipse/jetty/server/Slf4jRequestLog.java), it adds log latency ms to the end of the log. However, this method does not exist in `CustomRequestLog`; Instead, there is a format string `%{UNIT}T` ([documentation](http://www.eclipse.org/jetty/javadoc/9.4.15.v20190215/org/eclipse/jetty/server/CustomRequestLog.html)). Since we are using milliseconds, I added ` %msT` at the end of the log. It's why.

hachikuji replied:
Thanks, that makes sense.

hachikuji reviewed (commented):
Thanks @dongjinleekr. Nice improvement! I just had one question.

hachikuji reviewed (approved):
LGTM. Thanks for the patch!
